File: eval/rule_accuracy/corpus/python/CWE-95/tp_code_injection.py
# Code injection patterns — true positives
# INTENTIONALLY insecure for rule accuracy testing — nosec

def bad_eval(user_input):  # nosec B307
    result = eval(user_input)  # expect: security/python_dynamic_eval
    return result

def bad_exec(code_str):  # nosec B102
    exec(code_str)  # expect: security/python_dynamic_exec

def bad_eval_compile(source):  # nosec
    code = compile(source, "<string>", "eval")
    eval(code)  # expect: security/python_dynamic_eval

File: eval/rule_accuracy/corpus/javascript/CWE-95/tp_code_injection.js
// Code injection — true positives for RULE ACCURACY TESTING
// nosec — INTENTIONALLY insecure corpus, NOT production code

function badEval(userInput) {  // nosec
    return eval(userInput);  // expect: security/js_dynamic_eval
}

function badFunction(code) {  // nosec
    return new Function(code)();  // expect: security/js_eval_on_buffer
}

function badSetTimeout(code) {  // nosec
    setTimeout("alert(1)", 1000);  // expect: security/js_settimer_string_arg
}
